src/gui/folderstatusmodel.cpp

```diff
@@ -712,13 +712,17 @@ void FolderStatusModel::slotUpdateDirectories(const QStringList &list)
     if (!pathToRemove.endsWith(QLatin1Char('/')))
         pathToRemove += QLatin1Char('/');
 
-    QStringList selectiveSyncBlackList;
+    QSet<QString> selectiveSyncBlackList;
     bool ok1 = true;
     bool ok2 = true;
     if (parentInfo->_checked == Qt::PartiallyChecked) {
         selectiveSyncBlackList = parentInfo->_folder->journalDb()->getSelectiveSyncList(SyncJournalDb::SelectiveSyncBlackList, &ok1);
     }
-    auto selectiveSyncUndecidedList = parentInfo->_folder->journalDb()->getSelectiveSyncList(SyncJournalDb::SelectiveSyncUndecidedList, &ok2);
+    // qlist for historic reasons
```

Comment:
Such as?

Reply:
I didn't want to touch https://github.com/owncloud/client/pull/10685/files/d2bfee32e63eb81521e5e8c52b6793e81cfa16da#diff-b247ca89431ddfc0a0621c6a525d83a073fb9ff3b29ac6e68d1e571b5b9e6d06R810

Reply:
This link is broken.

Reply:
https://github.com/owncloud/client/pull/10685/files#diff-b247ca89431ddfc0a0621c6a525d83a073fb9ff3b29ac6e68d1e571b5b9e6d06R816

Reply:
I don't see a justification in this link. Could you add some more information to your comment, please?

Reply:
The algorithm would require a rewrite and I lack motivation for that.

Reply:
All I'm saying is that "historic reasons" does not make sense to me, probably because I don't know said history. Anyway.